src/test/fixtures/profiles/processed-profile.js

```diff
@@ -435,6 +412,36 @@ function _buildThreadFromTextOnlyStacks(
       // the indexes can double as both string indexes and func indexes.
       const funcIndex = stringTable.indexForString(funcName);
 
+      // Find the library name from the function name and create an entry if needed.
+      const libraryName = _findLibNameFromFuncName(funcNameWithModifier);
+      let resourceIndex = -1;
+      if (libraryName) {
+        resourceIndex = resourceIndexCache[libraryName];
```

Comment:
Just a nit, since we moved this code here, probably we can move the `resourceIndexCache` initialization to above the `const categoryOther = categories.findIndex(c => c.name === 'Other');` line to move it closer to the place we actually use that variable.